tests/RunnerTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace DepDocTest;
+
+use DepDoc\Application;
+use DepDoc\Runner;
+use PHPUnit\Framework\TestCase;
+
+class RunnerTest extends TestCase
+{
+    public function testRunOutputsHelp()
+    {
+        $runner = new Runner();
+
+        ob_start();
+        $exitCode = $runner->run(['-h']);
+
+        $output = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertContains('depdoc [command] [options]', $output);
+        $this->assertEquals(0, $exitCode);
+    }
+
+    public function testRunDefaultsToValidate()
+    {
+        $application = $this->prophesize(Application::class);
+        $application
+            ->validateAction()
+            ->willReturn(true);
+
+        $runner = new Runner();
+        $runner->setApplication($application->reveal());
+
+        $exitCode = $runner->run();
+        $this->assertEquals(0, $exitCode);
+    }
+
+    public function testRunUseProvidesArgument()
+    {
+        $application = $this->prophesize(Application::class);
+        $application
+            ->validateAction()
+            ->willReturn(true);
+        $application
+            ->updateAction()
+            ->willReturn(true);
```

Comment:
->shouldBeCalled();

Wäre noch sinnvoll, damit man auch sicher ist das die Methoden aufgerufen werden.

Reply:
@reinfi das checkt PHPUnit bzw Prophecy automatisch.
Jeder Mock einer Methode ist gleichzeitig ein "Versprechen", das diese so aufgerufen wird.
Ist dem nicht so, schlägt der Test fehl.

In früheren PHPUnit Versionen musste man das selber prüfen, da Prophecy nicht fest integriert war.
Jetzt passiert es quasi im tear down

Reply:
Das ist leider nicht richtig. Selbst mit PHPUnit 7.x ist es immer noch so, dass man jede beliebige Methode mocken kann, aber wenn sie auch aufgerufen werden soll man das explizit angeben muss.

Reply:
Du hast absolut recht, vielen Dank!
Hab das gerade getestet und ja, der Mock einer Methode erzeugt nicht automatisch eine `CallPrediction` - weiß nicht, wieso ich das bisher annahm. 👍